### ISSUE TITLE
Problem of accuracy

### ISSUE THREAD

**Issue by jajoe** [mljs/random-forest#7]
The accuracy seems to be worst than with Scikit-learn. 

<details> 
<summary>sklearn</summary>

```python
# coding: utf8

from sklearn.metrics import accuracy_score
from sklearn.ensemble import RandomForestClassifier

def do_RandomForest(x_train, y_train, x_test, y_test, n=1000):
  rf = RandomForestClassifier(n_estimators=n)
  rf.fit(x_train, y_train)
  score = rf.score(x_test, y_test)
  y_predict = rf.predict(x_test)
  return score, y_predict

if __name__ == "__main__":
    x_train = [[0,-1], [1,0], [1,1], [1,-1], [2,0], [2,1], [2,-1], [3,2], [0,4], [1,3], [1,4], [1,5], [2,3], [2,4], [2,5], [3,4], [1, 10], [1, 12], [2, 10], [2,11], [2, 14], [3, 11]]
    y_train = [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2]
    x_test = [[0, -2], [1, 0.5], [1.5, -1], [1, 2.5], [2, 3.5], [1.5, 4], [1, 10.5], [2.5, 10.5], [2, 11.5]]
    y_test = [0, 0, 0, 1, 1, 1, 2, 2, 2]
    score, y_star = do_RandomForest(x_train, y_train, x_test, y_test)
    print score
    print y_star
    # output : 
    # 1.0 
    # [0 0 0 1 1 1 2 2 2]
```
</details>

<details> 
<summary>ml-random-forest</summary>

```javascript
const {Matrix} = require('ml-matrix');
const {RandomForestClassifier} = require('ml-random-forest');

// our training set (X,Y)
var X = new Matrix([[0,-1], [1,0], [1,1], [1,-1], [2,0], [2,1], [2,-1], [3,2], [0,4], [1,3], [1,4], [1,5], [2,3], [2,4], [2,5], [3,4], [1, 10], [1, 12], [2, 10], [2,11], [2, 14], [3, 11]]);
var Y = Matrix.columnVector([0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2]);

// the test set (Xtest, Ytest)
var Xtest = new Matrix([[0, -2], [1, 0.5], [1.5, -1], [1, 2.5], [2, 3.5], [1.5, 4], [1, 10.5], [2.5, 10.5], [2, 11.5]])
var Ytest = Matrix.columnVector([0, 0, 0, 1, 1, 1, 2, 2, 2]);

// we will train our model
var rf = new RandomForestClassifier({nEstimators: 1000});
rf.train(X,Y);

// we try to predict the test set
var finalResults = rf.predict(Xtest);
console.log(finalResults)
// Now, you can compare finalResults with the Ytest, which is what you wanted to have.
// output (expected is [0, 0, 0, 1, 1, 1, 2, 2, 2]):
// [ 0, 0, 1, 0, 1, 1, 0, 2, 1 ]
// 5 true predictions (9 values) -> 55% accuracy 
```
</details>

I don't see where is the problem with ml-random-forest, but the results don't seem to be good...

**Comment by jajoe** [comment]
Maybe it was because of the Decision-Tree, but I've tested with the Decision-Tree and the results are good (better than with RF).

<details> 
<summary>ml-cart</summary>

```javascript
const {Matrix} = require('ml-matrix');
const {DecisionTreeClassifier} = require('ml-cart');

// our training set (X,Y)
var X = new Matrix([[0,-1], [1,0], [1,1], [1,-1], [2,0], [2,1], [2,-1], [3,2], [0,4], [1,3], [1,4], [1,5], [2,3], [2,4], [2,5], [3,4], [1, 10], [1, 12], [2, 10], [2,11], [2, 14], [3, 11]]);
var Y = Matrix.columnVector([0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2]);

// the test set (Xtest, Ytest)
var Xtest = new Matrix([[0, -2], [1, 0.5], [1.5, -1], [1, 2.5], [2, 3.5], [1.5, 4], [1, 10.5], [2.5, 10.5], [2, 11.5]])
var Ytest = Matrix.columnVector([0, 0, 0, 1, 1, 1, 2, 2, 2]);

// we will train our model
var dt = new DecisionTreeClassifier({minNumSamples: 3, maxDepth:10});
dt.train(X,Y);

// we try to predict the test set
var finalResults = dt.predict(Xtest);
console.log(finalResults)
// output (expected is [0, 0, 0, 1, 1, 1, 2, 2, 2]):
// [ 0, 0, 0, 1, 1, 1, 2, 2, 2 ]
// 100% accuracy
```
</details>
I believe that there is a problem in the source code of the RF.

**Comment by jajoe** [comment]
Now everything is working :-)

**Comment by metda** [comment]
hi Jajoe, 
was the accuracy accurate in your case?

Thanks.
